### PR TITLE
Update rewritings for qwen

### PR DESCRIPTION
**Pull request by RunningLeon**
## Motivation

update rewritings for the latest `modeling_qwen.py` as in [huggingface](https://huggingface.co/Qwen/Qwen-7B-Chat/blob/main/modeling_qwen.py)

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

None

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit tests to ensure the correctness.
3. If the modification has a dependency on downstream projects of a newer version, this PR should be tested with all supported versions of downstream projects.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: lmdeploy/pytorch/models/qwen.py [hunk 1]
```diff
@@ -215,21 +209,27 @@ def _continuous_batching_forward(
...
         hidden_states = inputs_embeds
 
-        # decoder layers
-        for idx, decoder_layer in enumerate(self.h):
-            past_key_value = (past_key_values[idx]
-                              if past_key_values is not None else None)
-            layer_outputs = decoder_layer(
+        context = self.context.context
+        max_kv_seq_length = context.max_kv_seq_length
+        # do not support use_dynamic_ntk
```
Comment (lvhan028) [inline lmdeploy/pytorch/models/qwen.py:214]: why?

Comment (RunningLeon) [reply lmdeploy/pytorch/models/qwen.py:214]: for these reasons, we could do not support this feature for now:
1. dynamic_ntk only works on prefilling phase where input_len is greater than `seq_lenth` in config.json, which is 8192 for qwen7b
2. dynamic_ntk should assign different ntk_alphas for different sequences in a batch, which is complicated to implement in pytorch engine
3. Qwen2 has remove `dynamic_ntk`, which means it is not so important.
4. Evaluations results with opencompass suggests that it won't change the results too much:

Comment (jjjjohnson) [reply lmdeploy/pytorch/models/qwen.py:214]: But seq_lenth for Qwen-14B is 2048
https://huggingface.co/Qwen/Qwen-14B/blob/main/config.json

Comment (RunningLeon) [reply lmdeploy/pytorch/models/qwen.py:214]: > But seq_lenth for Qwen-14B is 2048 https://huggingface.co/Qwen/Qwen-14B/blob/main/config.json

Yes. you are right. How do you use lmdeploy with Qwen-14B on pytorch engine? is dynamic ntk useful for you ? Why not use Qwen2?

**Review by lvhan028 — APPROVED** [review]

**Pull request merged**